modeshape-jcr/src/main/java/org/modeshape/connector/filesystem/FileSystemConnector.java

```diff
@@ -481,16 +481,30 @@ public boolean hasDocument( String id ) {
         return fileFor(id).exists();
     }
 
+    /**
+     * Hook for processing other node types (not resources or regular files)
+     * @param id
+     * @param file
+     * @param isRoot
+     * @return true if node is recognized and processed; false causes fallback
+     * to processing as a directory.
+     */
+    protected DocumentWriter otherType(String id, File file, boolean isRoot, boolean isResource) {
```

Comment:
what about renaming this to `writerFor(...)` ? Seems like a better name given the intent of the method